src/NuGet.Core/NuGet.Build.Tasks/RestoreTask.cs

```diff
@@ -140,11 +145,18 @@ private async Task<bool> ExecuteAsync(Common.ILogger log)
 
             EmbedInBinlog = GetFilesToEmbedInBinlog(dgFile);
 
+            if (RestoreNoCache)
```

Comment:
If we log this every time + implement https://github.com/dotnet/sdk/pull/35107, we're gonna end up warning twice in the dotnet.exe case. 

One simple way to fix it would be to use conditional compilation here, but we can also just log twice. I don't see that being a major concern.

Reply:
From my understanding, the dotnet sdk calls MSBuild with the property RestoreNoCache, as a result, I decided to not add a log message in the sdk implementation to avoid logging the same message twice. So, I think, we would just have the one warning from MSBuild. I am not sure where the second log message comes from.

Reply:
mb, I made some wrong assumptions there. You can ignore this.